recipes/magicblast/build.sh

```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+set -euxo pipefail
+
+mkdir -p $PREFIX/bin
+
+# the binary expects libbz1.so.1, but the available one is libbz2.so.1.0
+patchelf --replace-needed libbz2.so.1 libbz2.so.1.0 bin/magicblast
```

Comment:
This is hairy. If you must stick to the binaries and must absolutely patch the ELF, please add a test case that reads/writes bz2 files to make sure this actually works.

This also will not work on OSX. Different tools there.

Reply:
The comment is misleading: It’s not `libbz1.so.1`, but `libbz2.so.1` in both cases, except that the available has an additional `.0` in the end. The binary is linked against a more generic soname than it should be. And even then, libbz2 hasn’t changed in years, so there is really no need to add a test case. Also, with the patch, the binary runs. Without it, it doesn’t. So running the binary in the tests is enough. (I’m using the same patch in the IgBLAST recipe.)

Reply:
If I remember correctly, you can just skip the `patchelf` part on macOS, so wrap it in a `if [[ $(uname) == Linux ]]; then ... fi` and add a `# [linux]` selector to the patchelf dependency in the recipe.

Reply:
Thanks @marcelm I added changes that you suggested.

@epruesse, magicblast does not read/write any bz2 files directly. I added a test case that downloads data from NCBI. bz2 may be used there.